subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultSourceDirectorySet.java

```diff
@@ -164,17 +170,33 @@ public PatternFilterable getFilter() {
 
     @Override
     public File getOutputDir() {
-        return outputDir.get();
+        return destinationDirectory.getAsFile().get();
```

Comment:
This should have a deprecation nag, if it's deprecated.

Reply:
Will add in follow up deprecation PR